akka-stream-tests/src/test/java/akka/stream/javadsl/GraphDslTest.java

```diff
@@ -0,0 +1,97 @@
+        final Flow<Integer, Integer, NotUsed> f1 = Flow.of(Integer.class).map(elem -> elem + 10);
+        final Flow<Integer, Integer, NotUsed> f2 = Flow.of(Integer.class).map(elem -> elem + 20);
+        final Flow<Integer, String, NotUsed> f3 = Flow.of(Integer.class).map(elem -> elem.toString());
+        final Flow<Integer, Integer, NotUsed> f4 = Flow.of(Integer.class).map(elem -> elem + 30);
+
+        final RunnableGraph<CompletionStage<List<String>>> g =
+                RunnableGraph.fromGraph(
+                        GraphDSL     // create() function binds sink, out which is sink's out port and builder DSL
+                                .create(   // we need to reference out's shape in the builder DSL below (in to() function)
+                                        sink,                // previously created sink (Sink)
+                                        (builder, out) -> {  // variables: builder (GraphDSL.Builder) and out (SinkShape)
+                                            final UniformFanOutShape<Integer, Integer> bcast = builder.add(Broadcast.create(2));
+                                            final UniformFanInShape<Integer, Integer> merge = builder.add(Merge.create(2));
+
+                                            final Outlet<Integer> source = builder.add(in).out();
+                                            builder.from(source).via(builder.add(f1))
+                                                    .viaFanOut(bcast).via(builder.add(f2)).viaFanIn(merge)
+                                                    .via(builder.add(f3.grouped(1000))).to(out);  // to() expects a SinkShape
+                                            builder.from(bcast).via(builder.add(f4)).toFanIn(merge);
+                                            return ClosedShape.getInstance();
+                                        }));
+        CompletionStage<List<String>> result = g.run(materializer);
+        assert result.toCompletableFuture().get(1, TimeUnit.SECONDS).containsAll(Arrays.asList("31", "41", "32", "42", "33", "43", "44", "34", "45", "35"));
+    }
+
+    @Test
+    public void beAbleToConstructClosedGraphFromList() throws Exception {
+        final Source<String, NotUsed> in = Source.from(Arrays.asList("a", "b", "c"));
+
+        final Sink<String, CompletionStage<String>> sink1 = Sink.head();
+        final Sink<String, CompletionStage<String>> sink2 = Sink.head();
+        final Sink<String, CompletionStage<String>> sink3 = Sink.head();
+        final Sink<String, CompletionStage<String>> sink4 = Sink.head();
+
+        final List<Sink<String, CompletionStage<String>>> list = new ArrayList<>();
+        list.add(sink1);
+        list.add(sink2);
+        list.add(sink3);
+        list.add(sink4);
+
+        final RunnableGraph<List<CompletionStage<String>>> g = RunnableGraph.fromGraph(
+                GraphDSL.create(
+                                list,
+                                (builder, outs) -> {
```

Comment:
If you type the parameter list explicitly I think the "ugly cast" is not necessary any more.